jupyterhub/user.py

```diff
@@ -235,8 +235,11 @@ def spawn(self, options=None):
         api_token = self.new_api_token()
         db.commit()
 
+        self.server_base_url = base_url
+
         spawner = self.spawner
```

Comment:
yes. A Spawner corresponds to a server instance, so we have to have a Spawner instance for every running server.